doc/CMakeLists.txt

```diff
@@ -0,0 +1,39 @@
+# Provide a local project and cmake_minimum_required calls
+# to permit to generate the documentation without configuring
+# the complete project
+
+cmake_minimum_required(VERSION 3.0)
+
+# This variable should contain the argument of the project call in the main CMakeLists.txt
+set(ORIGINAL_PROJECT_NAME "LibTemplateCMake")
+project(${ORIGINAL_PROJECT_NAME}Docs)
+
+include(GNUInstallDirs)
+
+find_package(Doxygen QUIET)
+
+if(DOXYGEN_FOUND)
+  # This variables are used inside the Doxyfile.in template
+  set(DOXYGEN_BUILD_DIR "${CMAKE_CURRENT_BINARY_DIR}" CACHE STRING "Build directory for Doxygen html output (absolute).")
+  set(DOXYGEN_INSTALL_DIR "${CMAKE_INSTALL_DATAROOTDIR}/doc/${ORIGINAL_PROJECT_NAME}" CACHE STRING "Installation directory for Doxygen html output (relative to CMAKE_INSTALL_PREFIX).")
+  get_filename_component(ORIGINAL_PROJECT_DIR "${CMAKE_CURRENT_SOURCE_DIR}" PATH)
```

Comment:
I would change PATH with DIRECTORY. PATH is the legacy call for CMake 2.8 which we do not support.

Reply:
Good catch, I actually just copy&pasted from the files generated by `export(TARGETS ..)`